homeassistant/components/growatt_server/sensor.py

```diff
@@ -315,6 +315,30 @@ def get_data(self, entity_description):
                     "%s - No drop detected, using API value", entity_description.name
                 )
 
+        # Lifetime total values should always be increasing, they will never reset, however
+        # the API sometimes returns 0 values when the clock turns to 00:00 local time
+        # in that scenario we should just return the previous value
+        # Scenarios:
+        # 1 - System has a genuine 0 value when it it first commissioned:
+        #        - will return 0 until a non-zero value is registered
+        # 2 - System has been running fine but temporarily resets to 0 briefly at midnight:
+        #        - will return the previous value
+        # 3 - HA is restarted during the midnight 'outage' - Not handled:
```

Comment:
Maybe solve this by inheriting from RestoreSensor and restoring the last state when creating the entity?

https://github.com/home-assistant/core/blob/14ee3f1ddc7295e27242751fac5768abaae2e80c/homeassistant/components/sensor/__init__.py#L959

Reply:
Thanks for the suggestion I'll look into how I can do this - I was wondering if there's a nice way of getting the previous value of an existing sensor, clearly there is!!

If it's ok with you, I'll look to implement this in a future PR rather than this one as I'd like to get this one merged as-is.

Reply:
Yes. 👍